src/compile/axis/assemble.ts

```diff
@@ -158,6 +158,11 @@ export function assembleAxis(
       setAxisEncode(axis, 'labels', 'text', {signal: expr});
     }
 
+    // clear default label align so label flush still works
+    if (axis.labelAngle === 0 && (orient === 'top' || orient === 'bottom') && axis.labelAlign === 'center') {
+      delete axis.labelAlign;
```

Comment:
This is too hacky.  Setting it in parse and removing it later would make it really hard to reason about. 
Also, what if I make labelFlush = false, shouldn't we still set textAlign here?

, and wouldn't be correct in all cases. 

I'd rather make labelAlign/labelBaseline also depends on labelFlush. 
Basically if labelFlush is true, then don't set labelAlign for x / labelBaseline for y.

You can extract labelFlush to be AxisRuleParams, just like we do with labelAngle.

Reply:
I did that but when I returned `undefined` in the case where I didn't want to set it, then the resolution logic would fill in the wrong value. I could return `null` instead of undefined, I guess.

Reply:
The issue with `null` is that we need to clean it up later.

Reply:
I think cleaning up null is more reasonable than setting other values then delete?

At least, we can consistently set null for both signal and normal cases.  (And only clean up in normal cases.)